slick/src/main/scala/slick/lifted/ExtensionMethods.scala

```diff
@@ -58,6 +58,23 @@ trait ColumnExtensionMethods[B1, P1] extends Any with ExtensionMethods[B1, P1] {
     Library.IfNull.column[P2](n, e.toNode)(tpe(e))
 }
 
+final class Tuple2ColumnExtensionMethods[A: TypedType, B: TypedType](val c: Rep[(A, B)]) extends BaseExtensionMethods[(A, B)] {
+  def inSet[R](seq: Iterable[(A, B)])(implicit om: o#to[Boolean, R]) =
+    if(seq.isEmpty) om(LiteralColumn(false))
+    else {
+      val pairs = seq.map(v => ProductNode(ConstArray(LiteralNode(implicitly[TypedType[A]], v._1), LiteralNode(implicitly[TypedType[B]], v._2))))
```

Comment:
treat each element of the tuple as a literal in a product (rather than treating the whole tuple as a literal)